pkg/routeagent_driver/handlers/ovn/handler.go

```diff
@@ -0,0 +1,155 @@
+package ovn
+
+import (
+	"net"
+	"sync"
+
+	"github.com/pkg/errors"
+	"k8s.io/klog"
+
+	submV1 "github.com/submariner-io/submariner/pkg/apis/submariner.io/v1"
+	cableCleanup "github.com/submariner-io/submariner/pkg/cable/cleanup"
+	"github.com/submariner-io/submariner/pkg/cable/wireguard"
+	clientset "github.com/submariner-io/submariner/pkg/client/clientset/versioned"
+	"github.com/submariner-io/submariner/pkg/event"
+	"github.com/submariner-io/submariner/pkg/routeagent/cleanup"
+	"github.com/submariner-io/submariner/pkg/routeagent_driver/environment"
+	"github.com/submariner-io/submariner/pkg/util"
+)
+
+type Handler struct {
+	event.HandlerBase
+	mutex                 sync.Mutex
+	config                *environment.Specification
+	smClient              clientset.Interface
+	cableRoutingInterface *net.Interface
+	cleanupHandlers       []cleanup.Handler
+	localEndpoint         *submV1.Endpoint
+	remoteEndpoints       map[string]*submV1.Endpoint
+	isGateway             bool
+}
+
+func NewHandler(env environment.Specification, smClientSet clientset.Interface) *Handler {
+	return &Handler{
+		config:   &env,
+		smClient: smClientSet,
+	}
+}
+
+func (ovn *Handler) GetName() string {
+	return "ovn-hostroutes-handler"
+}
+
+func (ovn *Handler) GetNetworkPlugins() []string {
+	return []string{"OVNKubernetes"}
+}
+
+func (ovn *Handler) Init() error {
+	var err error
+
+	ovn.cableRoutingInterface, err = util.GetDefaultGatewayInterface()
+	if err != nil {
+		klog.Fatalf("Unable to find the default interface on host: %s", err.Error())
+	}
+
+	// For now we get all the cleanups
+	ovn.cleanupHandlers = cableCleanup.GetCleanupHandlers()
+
+	return nil
+}
+
+func (ovn *Handler) LocalEndpointCreated(endpoint *submV1.Endpoint) error {
+	ovn.mutex.Lock()
+	defer ovn.mutex.Unlock()
+
+	ovn.localEndpoint = endpoint
+
+	// TODO: this logic belongs to the cabledrivers instead
+	if endpoint.Spec.Backend == "wireguard" {
+		//NOTE: This assumes that LocalEndpointCreated happens before than TransitionToGatewayNode
+		if wg, err := net.InterfaceByName(wireguard.DefaultDeviceName); err == nil {
+			ovn.cableRoutingInterface = wg
```

Comment:
We are overriding the interface we discovered in the Init method. Can you please explain the significance of cableRoutingInterface? Is it related to cniInterface?

Reply:
It's the interface to which we forward to/from traffic for the remote clusters, it's used here:

https://github.com/submariner-io/submariner/pull/971/files#diff-f45ea0a8359161724c3b2f14cbfffa677d410ddf4a2856be29f42e1eb705b3e4R70

I will refactor this to make it clearer, when this event happens and we don't have an existing interface detected, I will make the detection. Also, only once, not multiple times like we would be doing now for wireguard.